src/Plutip/UtxoDistribution.purs

```diff
@@ -94,6 +94,23 @@ instance UtxoDistribution InitialUTxOsWithStakeKey KeyWallet where
         tail
   keyWallets _ wallet = [ wallet ]
 
+instance UtxoDistribution (Array InitialUTxOs) (Array KeyWallet) where
+  encodeDistribution amounts = amounts
+  decodeWallets _ privateKeyResponses =
+    Just $
+      ( \(PrivateKeyResponse key) -> privateKeysToKeyWallet
+          (PrivatePaymentKey key)
+          Nothing
+      ) <$> privateKeyResponses
+  decodeWallets' listOfInitialUTxOs privateKeyResponses =
+    let
+      wallets = Array.mapMaybe
```

Comment:
I doubt we should use mapMaybe here. If one decoding action fails, everything should fail. Otherwise the indices will all be messed up, right?

Reply:
Right, I replaced the logic to return `Nothing` if one of the decodings resutls in `Nothing`